datalad_next/patches/tests/test_push_to_export_remote.py

```diff
@@ -0,0 +1,81 @@
+from unittest.mock import patch
+
+from datalad.tests.utils import (
+    assert_false,
+    assert_in,
+    assert_in_results,
+    assert_raises,
+    assert_status,
+    assert_true,
+    eq_,
+    ok_,
+)
+
+
+from datalad_next.patches.push_to_export_remote import _is_export_remote
+
+
+class MockRepo:
+    def get_special_remotes(self):
+        return {
+            0: {
+                "name": "no-target",
+                "exporttree": "no"
+            },
+            1: {
+                "name": "yes-target",
+                "exporttree": "yes"
+            }
+        }
+
+
+def test_is_export_remote():
+    # Ensure that None is handled properly
+    assert_false(_is_export_remote(None))
+
+    # Ensure that dicts without "exporttree" keyword are handled correctly
+    assert_false(_is_export_remote({}))
+
+    # Ensure that "exporttree" is interpreted correctly
+    assert_false(_is_export_remote({"exporttree": "no"}))
+    assert_true(_is_export_remote({"exporttree": "yes"}))
+
+
+def test_patch_pass_through():
+    # Ensure that the original _transfer_data is called if the target remote
+    # has exporttree # not set to "yes"
+    from datalad.core.distributed.push import _transfer_data as _original_transfer_data
+    from datalad_next.patches.push_to_export_remote import _transfer_data
+
+    print(_original_transfer_data, _transfer_data)
+    _transfer_data(*([None]*9))
+    with patch("datalad_next.patches.push_to_export_remote.push._push_data") as pd_mock:
+        _original_transfer_data(
```

Comment:
@christian-monch
Took me a while to figure, but this test does fail b/c this call only creates the generator. It doesn't execute anything.

Reply:
Thx. (face-palm)